### PR TITLE
feat: Support process scoped variables

### LINKED ISSUE #361: Scope Variable to Runbook process
We need to scope OctopusDeploy variable to Runbook process (screenshot). It can be done only via OctopusDeploy UI. The provider does not support it at the moment.

E.g.:

```
resource "octopusdeploy_variable" "myvar" {
  ...
  scope {
    processes = ["Runbooks-123"]
  }

```

image

### PR DESCRIPTION
**Pull request by tleed5**
[SC-66829]
Based on https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/562 

# Result
- Updated `go-octopusdeploy` version to include the changes that allow for process scope variable matching (see https://github.com/OctopusDeploy/go-octopusdeploy/pull/230)
- Add `process` to variable scope schema to allow it to be used in terraform. 

# Example 
```
resource "octopusdeploy_variable" "test_variable" {
  owner_id  = "Projects-21"
  type      = "String"
  name      = "Test Process Scope"
  value     = "Scopey"
  scope {
    processes = ["Runbooks-1"]
  }
}
```

fixes #361

### CODE REVIEW

**Review by zentron — CHANGES_REQUESTED** [review]
Looks reasonable. If we can add (or update an existing) test to include the usage of this parameter would be nice.

File: terraform/49-variables/variables.tf
Comment (tleed5) [inline terraform/49-variables/variables.tf:1]: Tenant scoped variables are already been tested [here](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/78410f278967836377b4cb6fd361961abde38116/terraform/26-tenant_variables/tenant_variables.tf) so it didn't seem necessary to test them again

File: integration_test.go [hunk 1]
```diff
@@ -3350,3 +3350,30 @@ func TestK8sPodAuthTargetResource(t *testing.T) {
...
+		project, err := client.Projects.GetByName("Test")
+		variableSet, err := client.Variables.GetAll(project.ID)
+
+		if err != nil {
+			return err
+		}
+
+		if len(variableSet.Variables) != 7 {
+			t.Fatalf("Expected 7 variables to be created")
+		}
```
Comment (tleed5) [inline integration_test.go:3375]: I figure we can assert test that the correct amount of variables are created, anything else would be covered by the fact the variable resources are created correctly.

**Review by zentron — APPROVED** [review]
Apart from maybe a quick tweak to the test to validate that the scope exists as expected, even just explicitly checking some known value for each differently scoped variable then feel free to merge.
🙇 Thanks for pushing this one forward

**Pull request merged**